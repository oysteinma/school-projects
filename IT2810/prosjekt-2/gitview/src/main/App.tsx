import { HashRouter, Routes, Route } from "react-router-dom";
import { createContext, useState } from "react";
import { Navbar } from "./components/Menu";
import { Form } from "./components/Form";
import { User } from "./components/User";
import { Commit } from "./components/Commit";
import { Issue } from "./components/Issue";
import { Repository } from "./components/Repository";
import Footer from "./components/Footer";

export const AuthContext = createContext({
  authenticated: false,
  setAuthenticated: (auth: boolean) => {
    console.log(auth);
  },
});
/**
 *
 * App Gives a general structure for the website. on top
 * you have the navbar, then the respective page, then
 * the navbar.
 */
const App = () => {
  const [authenticated, setAuthenticated] = useState(
    window.localStorage.getItem("id") && window.sessionStorage.getItem("token")
      ? true
      : false
  );
  return (
    <AuthContext.Provider value={{ authenticated, setAuthenticated }}>
      <HashRouter>
        <Navbar />
        <div className="content">
          {window.localStorage.getItem("id") &&
            window.sessionStorage.getItem("token") && <Repository />}
          <Routes>
            <Route path="/" element={<Form />} />
            <Route path="/users" element={<User />} />
            <Route path="/commits" element={<Commit />} />
            <Route path="/issues" element={<Issue />} />
          </Routes>
        </div>
        <div className="footer">
          <Footer />
        </div>
      </HashRouter>
    </AuthContext.Provider>
  );
};

export default App;
