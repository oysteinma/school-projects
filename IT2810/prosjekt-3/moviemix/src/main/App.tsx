import { HashRouter, Routes, Route } from "react-router-dom";
import Navbar from "./components/Navbar";
import LandingPage from "./pages/LandingPage";
import YourRatingsPage from "./pages/YourRatings";
import SearchResults from "./pages/SearchResults";
import { makeVar, useMutation } from "@apollo/client";
import { CREATE_USER } from "./queries/query";

/**
 * Sets up ReactiveVar and creates a user if one does not exist
 * @returns HashRouter - the router for the application
 */

export const moviesVar = makeVar([]);
export default function App() {
  const [createUsers] = useMutation(CREATE_USER);

  if (window.localStorage.getItem("userId") === null) {
    createUsers({ variables: { input: [{}] } }).then((result) => {
      window.localStorage.setItem(
        "userId",
        result.data.createUsers.users[0].id
      );
    });
  }

  return (
    <HashRouter>
      <Navbar />
      <Routes>
        <Route path="/" element={<LandingPage />} />
        <Route path="/ratings" element={<YourRatingsPage />} />
        <Route path="/results" element={<SearchResults />} />
      </Routes>
    </HashRouter>
  );
}
