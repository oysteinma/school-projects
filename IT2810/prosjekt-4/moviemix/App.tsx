import { ApolloProvider } from "@apollo/client";
import { Provider } from "react-native-paper";
import { SafeAreaProvider } from "react-native-safe-area-context";
import { client } from "./config/client";
import SetupUser from "./index";
import Navigation from "./navigation";

/**
 * Starting point of the application. Wraps the app in the necessary providers.
 * @returns the application
 */
export default function App() {
  return (
    <ApolloProvider client={client}>
      <SafeAreaProvider>
        <Provider>
          <Navigation />
          <SetupUser />
        </Provider>
      </SafeAreaProvider>
    </ApolloProvider>
  );
}
