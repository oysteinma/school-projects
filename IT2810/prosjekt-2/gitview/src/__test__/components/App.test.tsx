import App from "../../main/App";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

it("check if App renders succesfully", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
