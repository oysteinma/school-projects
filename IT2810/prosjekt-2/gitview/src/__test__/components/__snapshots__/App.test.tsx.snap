// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if App renders succesfully 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-19sbeka-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
    >
      <img
        alt="logo"
        class="MuiBox-root css-v9aowo"
        src="logo.png"
      />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-146uix7-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Gitview
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="links MuiBox-root css-1w4tu4k"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-16jcpsg-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Users
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1cxzgv7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Commits
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1cxzgv7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Issues
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
  <div
    class="content"
  >
    <div
      id="form"
    >
      <div
        id="introform"
      >
        <h1>
          Welcome To GitView!
        </h1>
        <p>
          This is a simple website which lets you view the users, commits, and issues of your own GitLab repository. The pages lets you interact in different ways with the data, such that you can view more specific details about your own repository!
        </p>
        <hr />
        <p>
          Please enter your Project ID and Project Access Token down below to use the website.
        </p>
      </div>
      <div
        id="formdiv"
      >
        <div
          id="iddiv"
        >
          <input
            class="inputs"
            id="idinput"
            placeholder="Project ID"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          id="tokendiv"
        >
          <input
            class="inputs"
            id="tokeninput"
            placeholder="Project Token"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          id="buttondiv"
        >
          <button
            id="buttoninput"
            role="button"
            type="button"
          >
            Connect!
          </button>
        </div>
        <div
          id="gitdiv"
        >
          <img
            id="gitlogo"
            src="git-logo.png"
          />
        </div>
      </div>
      <div />
    </div>
  </div>
  <div
    class="footer"
  >
    <footer>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        style="background-color: white;"
      />
      <div
        class="MuiBox-root css-1n3rr7d"
      >
        GitView Â® 
        2022
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        style="background-color: white;"
      />
    </footer>
  </div>
</div>
`;
